v1/src/main/java/com/google/cloud/teleport/spanner/ddl/Index.java

```diff
@@ -179,6 +178,11 @@ private void prettyPrintGsql(Appendable appendable) throws IOException {
     if (interleaveIn() != null) {
       appendable.append(", INTERLEAVE IN ").append(quoteIdentifier(interleaveIn(), dialect()));
     }
+
+    if (!nullFiltered() && filter() != null && !filter().isEmpty()) {
+      appendable.append(" WHERE ").append(filter());
```

Comment:
This enables where clause for all type of indexes. Not just Search index. Correct?

Reply:
Yes, this enables it for all index types and is intentional.